File: orika-spring-boot-starter/src/main/java/net/rakugakibox/spring/boot/orika/OrikaMapperFactoryBuilderConfigurer.java
package net.rakugakibox.spring.boot.orika;

import ma.glasnost.orika.impl.DefaultMapperFactory.MapperFactoryBuilder;

/**
 * The interface to configure {@link MapperFactoryBuilder}.
 */
public interface OrikaMapperFactoryBuilderConfigurer {

    /**
     * Configures the {@link MapperFactoryBuilder}.
     *
     * @param orikaMapperFactoryBuilder the {@link MapperFactoryBuilder}.
     */
    void configure(MapperFactoryBuilder<?, ?> orikaMapperFactoryBuilder);

}

File: core/src/main/kotlin/dev/akkinoc/spring/boot/orika/OrikaMapperFactoryBuilderConfigurer.kt
package dev.akkinoc.spring.boot.orika

import ma.glasnost.orika.impl.DefaultMapperFactory.MapperFactoryBuilder

/**
 * The interface to configure [MapperFactoryBuilder].
 */
interface OrikaMapperFactoryBuilderConfigurer {

    /**
     * Configures the [MapperFactoryBuilder].
     *
     * @param orikaMapperFactoryBuilder The [MapperFactoryBuilder].
     */
    fun configure(orikaMapperFactoryBuilder: MapperFactoryBuilder<*, *>)

}
